test-suite/shared/src/test/scala/org/scalajs/testsuite/compiler/ArrayTest.scala

```diff
@@ -45,4 +45,15 @@ class ArrayTest {
     assertThrows(classOf[ArrayIndexOutOfBoundsException], a(Int.MaxValue) = 1)
   }
 
+  @Test
+  def arraySelectSideEffecting_issue_3848(): Unit = {
+    assumeTrue("Assuming compliant ArrayIndexOutOfBounds",
+      hasCompliantArrayIndexOutOfBounds)
```

Comment:
Ah! The second line was supposed to stay where it was, to have a 4-space indentation :p

Reply:
Oh dear :P